### PR TITLE
Cleanup: WC_Query

### PR DESCRIPTION
**Pull request by Ninos**
Since WP 4.5 `search_post_excerpt()` in `WC_Query()` is not needed anymore because post_excerpt is already added to the wp search query (see https://github.com/WordPress/WordPress/commit/6d8a3f3612c0a3993dc2284f95cddc5a786ee307).
`search_post_excerpt()` just leads to problems with using custom join queries which are rejoining into wp_posts, because added post_excerpt in `search_post_excerpt()` does not use a table prefix -> you get following error:
`[Column 'post_excerpt' in where clause is ambiguous`

We could fix this bug by adding `$wpdb->prefix` to the `preg_replace()`-function in `search_post_excerpt()`, but I think it's ok removing this part of code (WP 4.5 is not so young).

Relevant code changes:
https://github.com/WordPress/WordPress/blob/4.4-branch/wp-includes/query.php#L2196
https://github.com/WordPress/WordPress/blob/4.5-branch/wp-includes/query.php#L2159

Relevant commit:
https://github.com/WordPress/WordPress/commit/6d8a3f3612c0a3993dc2284f95cddc5a786ee307

### CODE REVIEW

**Comment by mikejolley** [comment]
You cannot remove public methods; leave them and deprecate.

**Comment by Ninos** [comment]
Yes, sry. I hope it's fine now :-)

File: includes/class-wc-query.php [hunk 1]
```diff
@@ -329,34 +329,16 @@ public function pre_get_posts( $q ) {
...
 		$this->remove_product_query();
 	}
 
 	/**
 	 * Search post excerpt.
-	 *
-	 * @access public
-	 * @param string $where (default: '')
-	 * @return string (modified where clause)
+	 * @deprecated 3.2.0 - Bot needed anymore since WordPress 4.5.
```
Comment (partounian) [inline includes/class-wc-query.php]: Not needed anymore*

**Comment by mikejolley** [comment]
We have not dropped 4.4 support yet. @claudiosanches thoughts?

**Comment by claudiosanches** [comment]
I vote to close it, if does not offer support for 4.4.

**Comment by Ninos** [comment]
Before 4.5 woocommerce just added this routine to add excerpts for search query, but that was not default prior Wordpress 4.5. So this will Not break much. If wished I can add a check for WP < 4.5.

PS: When is it planned to remove WP 4.4 support? And why are you guys supporting such old versions but not older major versions of woocommerce?

**Comment by claudiosanches** [comment]
@Ninos because old WP versions still receive patches.

**Comment by Ninos** [comment]
@claudiosanches there's no LTS support for old WP versions. Only some important security issues gets backported if nessessary :-)
https://codex.wordpress.org/Supported_Versions

**Comment by claudiosanches** [comment]
@Ninos yes, we know about it.

**Comment by mikejolley** [comment]
It doesn't break 4.4 just removes a feature. So I think it's fine.

**Pull request merged**